compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -617,7 +617,10 @@ impl<'a> Linker for GccLinker<'a> {
         match strip {
             Strip::None => {}
             Strip::Debuginfo => {
-                self.linker_arg("--strip-debug");
+                // illumos does not support --strip-debug
+                if self.sess.target.os != "illumos" {
+                    self.linker_arg("--strip-debug");
+                }
```

Comment:
I _believe_ the intended solution here would be to invoke an external tool after linking that could do this, similar to the handling for macOS as introduced in https://github.com/rust-lang/rust/pull/82037.

You probably could just reuse the code path there, for the time being. That way when somebody gets to fixing the way we represent linkers in rustc, illumos will get automatically handled the mostly “right” way alongside with macOS (though you should still keep the comment to the effect that illumos does support `--strip-all` and could be improved to use that too)

Reply:
Thanks for taking a look at this. I've pushed a new commit which uses the linker's `--strip-all` but invokes an external `strip` utility for the `--strip-debug` case. After talking to people who know more about linkers than I do, they advised that it is best to let the linker do this when possible, hence the hybrid approach.
It may be that the illumos linker can be extended in the future to support `--strip-debug` but until then this seems like the best approach, and I'd like people to be able to successfully run the tests on illumos again.